kinto_client/__init__.py

```diff
@@ -117,265 +110,113 @@ def request(self, method, url, data=None, permissions=None, **kwargs):
-            except KintoException as e:
-                if method == 'get' and e.response.status_code == 403:
-                    exception = BucketNotFound(name)
-                    exception.response = e.response
-                    exception.request = e.request
-                    raise exception
-                else:
-                    raise
-
-            self.data = info['data']
-            self.permissions = Permissions(
-                object='bucket',
-                permissions=info['permissions'])
-
-    def _get_collection_uri(self, collection_id):
-        return '%s/collections/%s' % (self.uri, collection_id)
-
-    def get_collection(self, name, **kwargs):
-        return Collection(name, bucket=self, session=self.session, **kwargs)
-
-    def list_collections(self):
-        uri = "%s/%s" % (self.uri, 'collections')
-        resp, _ = self.session.request('get', uri)
-
-        return [collection[ID_FIELD] for collection in resp['data']]
-
-    def create_collection(self, name, permissions=None):
-        return Collection(name, bucket=self, session=self.session,
-                          create=True, permissions=permissions)
-
-    def delete_collection(self, name):
-        uri = self._get_collection_uri(name)
-        resp, _ = self.session.request('delete', uri)
+        self._bucket_name = bucket
+        self._collection_name = collection
+        self.endpoints = Endpoints()
+
+    def _get_endpoint(self, name, bucket=None, collection=None, id=None):
+        kwargs = {
+            'bucket': bucket or self._bucket_name,
+            'collection': collection or self._collection_name,
+            'id': id
+        }
+        return self.endpoints.get(name, **kwargs)
+
+    # Buckets
+
+    def update_bucket(self, bucket=None, permissions=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        resp, _ = self.session.request('put', endpoint,
+                                       permissions=permissions)
+        return resp
+
+    def delete_bucket(self, bucket=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        resp, _ = self.session.request('delete', endpoint)
         return resp['data']
 
-    def save(self):
-        self.session.request('patch', self.uri, data=self.data,
-                             permissions=self.permissions)
-
-    def delete(self):
-        resp, _ = self.session.request('delete', self.uri)
-        return resp['data']
+    def get_bucket(self, bucket=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        try:
+            resp, _ = self.session.request('get', endpoint)
+        except KintoException as e:
+            exception = BucketNotFound(bucket or self._bucket_name)
+            exception.response = e.response
+            exception.request = e.request
+            raise exception
+        return resp
 
+    def create_bucket(self, *args, **kwargs):
```

Comment:
I am not sure if Python is capable of handling: ``create_bucket = update_bucket`` at a class level.